File: src/test/javascript/toRoman.js
var _ = require('ramda');

var symbolValues = [
    ["X", 10],
    ["V", 5],
    ["IV", 4],
    ["I", 1],
    ["", 0]
];

module.exports = function toRoman(arabic) {
    if (arabic === 0) { return ""; }

    var symbolValue = _.find(function (x) { return arabic >= x[1]; }, symbolValues);

    return symbolValue[0] + toRoman(arabic - symbolValue[1]);
}

File: src/test/scala/ToRoman.scala
object ToRoman {

  val symbolValues = Map(
    "X" -> 10,
    "V" -> 5,
    "IV" -> 4,
    "I" -> 1)

  def toRoman(arabic: Int): String = {
    findClosestSymbolValue(arabic) match {
      case (symbol, 0) => symbol
      case (symbol, value) => symbol + toRoman(arabic - value)
    }
  }

  def findClosestSymbolValue(value: Int): (String, Int) = {
    symbolValues.find { case (_, symbolValue) => value >= symbolValue } match {
      case Some(x) => x
      case None => ("", 0)
    }
  }


}
